code/modules/recycling/disposal.dm

```diff
@@ -118,64 +137,114 @@
 		var/obj/item/storage/S = I
 		if(!S.removal_allowed_check(user))
 			return
+
 		if((S.allow_quick_empty || S.allow_quick_gather) && length(S.contents))
 			S.hide_from(user)
-			user.visible_message("[user] empties \the [S] into \the [src].", "You empty \the [S] into \the [src].")
+			user.visible_message(
+				"<span class='notice'>[user] empties [S] into the disposal unit.</span>",
+				"<span class='notice'>You empty [S] into disposal unit.</span>",
+				"<span class='notice'>You hear someone emptying something into a disposal unit.</span>"
+				)
 			for(var/obj/item/O in S.contents)
 				S.remove_from_storage(O, src)
 			S.update_icon() // For content-sensitive icons
 			update()
 			return
 
+	// Borg using their gripper to throw stuff away.
+	if(istype(I, /obj/item/gripper/))
+		var/obj/item/gripper/gripper = I
+		// Gripper is empty.
+		if(!gripper.gripped_item)
+			to_chat(user, "<span class='warning'>There's nothing in your gripper to throw away!</span>")
+			return
+
+		gripper.gripped_item.forceMove(src)
+		user.visible_message(
+			"<span class='notice'>[user] places [gripper.gripped_item] into the disposal unit.</span>", 
+			"<span class='notice'>You place [gripper.gripped_item] into the disposal unit.</span>",
+			"<span class='notice'>You hear someone dropping something into a disposal unit.</span>"
+			)
+		return
+
+	// Someone has a mob in a grab.
 	var/obj/item/grab/G = I
-	if(istype(G))	// handle grabbed mob
-		if(ismob(G.affecting))
-			var/mob/GM = G.affecting
-			for(var/mob/V in viewers(usr))
-				V.show_message("[usr] starts putting [GM] into the disposal.", 3)
-			if(do_after(usr, 20, target = GM))
-				GM.forceMove(src)
-				for(var/mob/C in viewers(src))
-					C.show_message("<span class='warning'>[GM] has been placed in [src] by [user].</span>", 3)
-				qdel(G)
-				add_attack_logs(usr, GM, "Disposal'ed", !!GM.ckey ? null : ATKLOG_ALL)
+	if(istype(G))
+		// If there's not actually a mob in the grab, stop it. Get some help.
+		if(!ismob(G.affecting))
+			return
+
+		var/mob/GM = G.affecting
+		user.visible_message(
+			"<span class='warning'>[user] starts stuffing [GM] into the disposal unit!</span>",
+			"<span class='warning'>You start stuffing [GM] into the disposal unit.</span>",
+			"<span class='warning'>You hear someone trying to stuff someone else into a disposal unit!</span>"
+			)
+
+		// Abort if the target manages to scurry away.
+		if(!do_after(user, 20, target = GM))
+			return
+
+		GM.forceMove(src)
+		user.visible_message(
+			"<span class='warning'>[GM] has been stuffed into the disposal unit by [user]!</span>",
+			"<span class='warning'>You stuff [GM] into the disposal unit.</span>",
+			"<span class='warning'>You hear someone being stuffed into a disposal unit!</span>"
+			)
```

Comment:
Please unindent these trailing parentheses to be in line with the visible_message call

Reply:
Done.